
# HG changeset patch
# User tiwai
# Date 1195222828 -3600
# Node ID 97fb834c9b22528f9413c6aeab2075f83e0fcaaa
# Parent 1e32bc870695923a64f623c7c70b57284765ae1a
snd hda suspend latency: shorten codec read

not sleeping for every codec read/write but doing a short udelay and
a conditional reschedule has cut suspend+resume latency by about 1
second on my T60.

Signed-off-by: Ingo Molnar <mingo@elte.hu>

--- a/sound/pci/hda/hda_intel.c	Fri Nov 16 11:06:30 2007 +0100
+++ b/sound/pci/hda/hda_intel.c	Fri Nov 16 15:20:28 2007 +0100
@@ -561,7 +561,8 @@ static unsigned int azx_rirb_get_respons
 		}
 		if (!chip->rirb.cmds)
 			return chip->rirb.res; /* the last value */
-		schedule_timeout_uninterruptible(1);
+		udelay(10);
+		cond_resched();
 	} while (time_after_eq(timeout, jiffies));
 
 	if (chip->msi) {

